designs/2024-baseline-support/README.md

```diff
@@ -47,161 +47,191 @@ The baseline files includes details about the file where the error found, the ru
+- Generating the suppressions file can be based on the cache file and should be faster when the cache file is used.
+- Allows developers to re-generate the suppressions file or even adjust it manually and re-lint still taking the same cache into consideration.
+- It even allows developers to delete the suppressions file and still take advantage of the cached file in subsequent runs. 
 
-- Generating the baseline can be based on the cache file and should be faster when the cache file is used.
-- Allows developers to re-generate the baseline or even adjust it manually and re-lint still taking the same cache into consideration.
-- It even allows developers to delete the baseline and still take advantage of the cached file in subsequent runs. 
+# Implementation notes
 
-### Implementation Details
+To introduce the above-mentioned options, we will need to: 
 
-First of all we need to introduce a new type to hold the individual baseline result:
+* add the new options in `default-cli-options.js`.
+* adjust the config for optionator.
+* add the new options as comments and arguments for eslint.
+* update documentation to explain the newly introduced option
+
+A new type must be created:
 
 ``` js
 /**
- * @typedef {Record<string, { count: number }>} BaselineResult
+ * @typedef {Record<string, { count: number }>} SuppressedViolation
  */
 ```
 
-One way to approach this is to introduce a Manager class handling the baseline results - in particular, both generating a baseline and validating the results against a baseline.
-
 ``` js
-class BaselineResultManager {
+class SuppressedViolationsManager {
     /**
-     * Creates a new instance of BaselineResultManager.
-     * @param {string} baselineLocation The location of the baseline file.
+     * Creates a new instance of SuppressedViolationsManager.
+     * @param {string} suppressionsLocation The location of the suppressions file.
      */
-    constructor(baselineLocation) {}
+    constructor(suppressionsLocation) {}
 
     /**
-     * Generates the baseline from the provided lint results.
+     * Updates the suppressions file based on the current violations
+     * 
      * @param {LintResult[]} results The lint results.
-     * @returns BaselineResult[]
+     * @returns SuppressedViolation[]
      */
-    generateBaseline(results) 
+    suppressAll(results) 
 
     /**
-     * Checks the provided baseline results against the lint results.
-     * It filters and returns the lint results that are not in the baseline.
-     * It also returns the unmatched baseline results.
+     * Updates the suppressions file based on the current violations and the provided rule
      * 
      * @param {LintResult[]} results The lint results.
-     * @param {BaselineResult[]} baseline The baseline.
+     * @param {string} rule The rule to suppress.
+     */
+    suppressByRule(results, rule)
```

Comment:
In order to allow multiple rules to be suppressed, maybe we could allow an array as the second argument?

```suggestion
    suppressRules(results, rules)
```